### PR TITLE
BUG: de2 vefi load

### PR DESCRIPTION
**Pull request by jklenzing**
# Description

Addresses #211

Deprecates the combined vefimagb data product under de2_vefi, replaces with new `de2_vefimagb` instrument module.

This product contains multiple time indices, the new instrument allows users to choose which one to load. For the deprecated version, only the mag data is kept to tentatively solve the issue raised in #211.

# Type of change

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- This change requires a documentation update

# How Has This Been Tested?

```
    import datetime as dt
    import pysat

    # Set electric field instrument
    vefi = pysat.Instrument(platform='de2', name='vefimagb', tag='e')
    vefi.download(dt.datetime(1983, 1, 1))

    # Set magnetic field instrument
    mag = pysat.Instrument(platform='de2', name='vefimagb', tag='b')

    # Both instruments can be loaded from the same download
    vefi.load(1983, 1)
    mag.load(1983, 1)
```

**Test Configuration**:
* Operating system: Ventura
* Version number: Python 3.10.9

# Checklist:

- [x] Make sure you are merging into the ``develop`` (not ``main``) branch
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] Add a note to ``CHANGELOG.md``, summarizing the changes
- [x] Update zenodo.json file for new code contributors

If this is a release PR, replace the first item of the above checklist with the release
checklist on the wiki: https://github.com/pysat/pysat/wiki/Checklist-for-Release

### CODE REVIEW

File: pysatNASA/instruments/de2_vefi.py [hunk 1]
```diff
@@ -89,10 +105,60 @@
...
+    ----------
+    fnames : array-like
+        Iterable of filename strings, full path, to data files to be loaded.
+        This input is nominally provided by pysat itself.
+    tag : str
+        Tag name used to identify particular data set to be loaded.
+        This input is nominally provided by pysat itself. (default='')
+    inst_id : str
+        Instrument ID used to identify particular data set to be loaded.
+        This input is nominally provided by pysat itself. (default='')
```
Comment (jklenzing) [inline pysatNASA/instruments/de2_vefi.py:126]:
```suggestion
    inst_id : str
        Instrument ID used to identify particular data set to be loaded.
        This input is nominally provided by pysat itself. (default='')
    kwargs : dict
        Additional kwargs passed through to the instrument.
```

**Comment by jklenzing** [comment]
> 
> It looks like there needs to be some data cleaning implemented, since this is supposedly 'clean'.

or maybe a warning? Cleaning is only based on fill vals.

**Comment by aburrell** [comment]
Looking at the data some of those are clearly fill values, they just aren't recognized by the clean method.

**Comment by jklenzing** [comment]
> Looking at the data some of those are clearly fill values, they just aren't recognized by the clean method.

The fill vals are improperly set. Adding a warning.

File: pysatNASA/instruments/de2_vefimagb.py [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+
+
+# ----------------------------------------------------------------------------
+# Instrument methods
+
+# Use standard init routine
+init = functools.partial(mm_nasa.init, module=mm_de2, name='vefi')
+
+# No cleaning, use standard warning function instead
+clean = mm_nasa.clean_warn
```
[the remark this answers is not in the document]
Comment (jklenzing) [reply pysatNASA/instruments/de2_vefimagb.py:87]: It's a standard function in the package that warns people that data cleaning is not implemented

**Review by aburrell — APPROVED** [review]

**Pull request merged**